### PR TITLE
Add logic to ensure url has a valid https protocol

### PR DESCRIPTION
**Pull request by david-livefront**
## 🎟️ Tracking

N/A

## 📔 Objective

This PR updates the Url parsing logic to inject a valid protocol if none is present.

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed
  issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: app/src/main/java/com/x8bit/bitwarden/data/autofill/accessibility/parser/AccessibilityParserImpl.kt [hunk 1]
```diff
@@ -29,10 +31,19 @@ class AccessibilityParserImpl : AccessibilityParser {
                 rootNode
                     .findAccessibilityNodeInfosByViewId("$packageName:id/$viewId")
                     .map { accessibilityNodeInfo ->
-                        browser.urlExtractor(accessibilityNodeInfo.text.toString())
+                        browser
+                            .urlExtractor(accessibilityNodeInfo.text.toString())
+                            ?.trim()
+                            ?.let { rawUrl ->
+                                if (rawUrl.contains(other = ".") && !rawUrl.hasHttpProtocol()) {
```
Comment (dseverns-livefront) [inline app/src/main/java/com/x8bit/bitwarden/data/autofill/accessibility/parser/AccessibilityParserImpl.kt:38]: the `.` check is just to make sure that the URL has a valid domain following pattern of `foo.bar/whatever` ?

**Review by dseverns-livefront — APPROVED** [review]
One question for my own sanity, but looks good!

File: app/src/main/java/com/x8bit/bitwarden/data/autofill/accessibility/parser/AccessibilityParserImpl.kt (hunk 1 above)
Comment (david-livefront) [reply app/src/main/java/com/x8bit/bitwarden/data/autofill/accessibility/parser/AccessibilityParserImpl.kt:38]: That is my assumption, but honestly, I am not 100% sure. This logic was based on the code [here](https://github.com/bitwarden/mobile/blob/main/src/App/Platforms/Android/Accessibility/AccessibilityHelpers.cs#L425)

**Comment by david-livefront** [comment]
Thanks @dseverns-livefront

**Pull request merged**